framework/src/variables/MooseVariableBase.C

```diff
@@ -107,8 +111,10 @@ MooseVariableBase::MooseVariableBase(const InputParameters & parameters)
     paramError("family", "finite volume (fv=true) variables must be have MONOMIAL family");
   if (getParam<bool>("fv") && _fe_type.order != 0)
     paramError("order", "finite volume (fv=true) variables currently support CONST order only");
-
-  if (_count > 1)
+  bool is_array_variable = getParam<bool>("array_variable");
+  if (!is_array_variable && _count > 1)
+    paramError("array_variable", "array_variable must be true if 'component' is greater than one");
+  if (is_array_variable)
```

Comment:
```suggestion
  const auto is_array = getParam<bool>("array");
  if (_count > 1)
    mooseAssert(is_array, "Must be true with component > 1");
  if (is_array)
```

We moved better error checking to SystemBase, so this can just be an assertion